src/node/element/path/data.rs

```diff
@@ -185,52 +185,71 @@ impl<'l> Parser<'l> {
-            'v' => VerticalLine(Relative, parameters),
+            'V' => VerticalLine(Absolute, self.read_parameters(false)?.into()),
+            'v' => VerticalLine(Relative, self.read_parameters(false)?.into()),
 
-            'Q' => QuadraticCurve(Absolute, parameters),
-            'q' => QuadraticCurve(Relative, parameters),
+            'Q' => QuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            'q' => QuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'T' => SmoothQuadraticCurve(Absolute, parameters),
-            't' => SmoothQuadraticCurve(Relative, parameters),
+            'T' => SmoothQuadraticCurve(Absolute, self.read_parameters(false)?.into()),
+            't' => SmoothQuadraticCurve(Relative, self.read_parameters(false)?.into()),
 
-            'C' => CubicCurve(Absolute, parameters),
-            'c' => CubicCurve(Relative, parameters),
+            'C' => CubicCurve(Absolute, self.read_parameters(false)?.into()),
+            'c' => CubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'S' => SmoothCubicCurve(Absolute, parameters),
-            's' => SmoothCubicCurve(Relative, parameters),
+            'S' => SmoothCubicCurve(Absolute, self.read_parameters(false)?.into()),
+            's' => SmoothCubicCurve(Relative, self.read_parameters(false)?.into()),
 
-            'A' => EllipticalArc(Absolute, parameters),
-            'a' => EllipticalArc(Relative, parameters),
+            'A' => EllipticalArc(Absolute, self.read_parameters(true)?.into()),
+            'a' => EllipticalArc(Relative, self.read_parameters(true)?.into()),
 
             'Z' | 'z' => Close,
 
             _ => raise!(self, "found an unknown path command '{}'", name),
         }))
     }
 
-    fn read_parameters(&mut self) -> Result<Vec<Number>> {
+    fn read_parameters(&mut self, is_arc: bool) -> Result<Vec<Number>> {
         let mut parameters = Vec::new();
+        let mut cmd_index: usize = 0;
 
-        while let Some(number) = self.read_number()? {
+        while let Some(number) = self.read_number_or_flag(if is_arc { Some(cmd_index % 7) } else { None })? {
+            cmd_index = cmd_index + 1;
             parameters.push(number);
             self.reader.consume_whitespace();
             self.reader.consume_char(',');
         }
+
         Ok(parameters)
     }
 
+    fn read_number_or_flag(&mut self, arc_relative_index: Option<usize>) -> Result<Option<Number>> {
+        match arc_relative_index {
+            Some(x) if x == 3 || x == 4 => self.read_flag(),
+            _ => self.read_number()
+        }
+    }
+
+    fn read_flag(&mut self) -> Result<Option<Number>> {
+        self.reader.consume_whitespace();
+        let val = self.reader.next();
+        match val {
+            Some('0') => Ok(Some(0.0)),
+            Some('1') => Ok(Some(1.0)),
+            _ => raise!(self, "failed to parse flag param in arc")
```

Comment:
Let’s change it to “failed to parse a flag in an elliptical arc”.

Reply:
done